dragonchain/lib/database/redisearch.py

```diff
@@ -299,23 +304,64 @@ def generate_indexes_if_necessary() -> None:
     """Initialize redisearch with necessary indexes and fill them from storage if migration has not been marked as complete"""
     redisearch_redis_client = _get_redisearch_index_client("").redis
     needs_generation = not bool(redisearch_redis_client.get(INDEX_GENERATION_KEY))
+    needs_l5_generation = not bool(redisearch_redis_client.get(INDEX_L5_VERIFICATION_GENERATION_KEY))
     # No-op if indexes are marked as already generated
-    if not needs_generation:
+    if not needs_generation and not needs_l5_generation:
         return
-    # Create block index
-    _log.info("Creating block indexes")
-    _generate_block_indexes()
-    # Create indexes for transactions
-    _log.info("Creating transaction indexes")
-    _generate_transaction_indexes()
-    # Create smart contract index
-    _log.info("Creating smart contract indexes")
-    _generate_smart_contract_indexes()
-    # Mark index generation as complete
-    _log.info("Marking redisearch index generation complete")
-    redisearch_redis_client.delete(BLOCK_MIGRATION_KEY)
-    redisearch_redis_client.delete(TXN_MIGRATION_KEY)
-    redisearch_redis_client.set(INDEX_GENERATION_KEY, "a")
+
+    if needs_l5_generation:
+        # Create L5 verification indexes
+        _generate_l5_verification_indexes()
+        # Mark index generation as complete
+        redisearch_redis_client.delete(L5_BLOCK_MIGRATION_KEY)
+        redisearch_redis_client.set(INDEX_L5_VERIFICATION_GENERATION_KEY, "a")
+
+    if needs_generation:
+        # Create block index
+        _log.info("Creating block indexes")
+        _generate_block_indexes()
+        # Create indexes for transactions
+        _log.info("Creating transaction indexes")
+        _generate_transaction_indexes()
+        # Create smart contract index
+        _log.info("Creating smart contract indexes")
+        _generate_smart_contract_indexes()
+        # Mark index generation as complete
+        _log.info("Marking redisearch index generation complete")
+        redisearch_redis_client.delete(BLOCK_MIGRATION_KEY)
+        redisearch_redis_client.delete(TXN_MIGRATION_KEY)
+        redisearch_redis_client.set(INDEX_GENERATION_KEY, "a")
+
+
+def _generate_l5_verification_indexes() -> None:
+    client = _get_redisearch_index_client(Indexes.verification.value)
+    try:
+        client.create_index(
+            [
+                redisearch.NumericField("block_id", sortable=True),
+                redisearch.NumericField("prev_id", sortable=True),
+                redisearch.NumericField("timestamp", sortable=True),
+                redisearch.TextField("dc_id"),
```

Comment:
What's the usecase for this field? Textfields are sharded so that you can search with smaller tokens, but I'm assuming you only need this index because you will only ever search on it with the full dc_id, and not part of one.

This should probably be a tagfield in that case because they are lighter than textfields for both storage and querying.

Reply:
thanks man. Thats a good point. I only needed that one extra field so good call